picknik_reset_fault_controller/src/picknik_reset_fault_controller.cpp

```diff
@@ -62,11 +62,15 @@ CallbackReturn PicknikResetFaultController::on_init() { return CallbackReturn::S
 controller_interface::return_type PicknikResetFaultController::update(
   const rclcpp::Time & /*time*/, const rclcpp::Duration & /*period*/)
 {
-  if (realtime_publisher_ && realtime_publisher_->trylock())
+  if (realtime_publisher_)
   {
-    state_.data = static_cast<bool>(state_interfaces_[StateInterfaces::IN_FAULT].get_value());
-    realtime_publisher_->msg_.data = state_.data;
-    realtime_publisher_->unlockAndPublish();
+    auto state_op =
+      command_interfaces_[CommandInterfaces::RESET_FAULT_ASYNC_SUCCESS].get_optional();
+    if (state_op.has_value())
+    {
+      state_.data = static_cast<bool>(state_op.value());
+    }
+    realtime_publisher_->try_publish(state_);
```

Comment:
`try_publish` is available in Jazzy and newer, correct?

In a separate PR we should update the readme and the table of supported versions , branches, build statuses, etc to reflect this.

Reply:
ad 2: When some new API is backported, it might be that there is no hint in the migration notes about that (nothing to migrate because we don't remove/break API). The [release notes](https://control.ros.org/rolling/doc/release_notes/release_notes.html) still should reflect this.

I'd recommend option 4, as it is the quickest one. And add proper CI jobs for all target distros ;)